omegaconf/grammar/OmegaConfGrammarParser.g4

```diff
@@ -70,7 +77,7 @@ primitive:
 
 // Same as `primitive` except that `COLON` and interpolations are not allowed.
 dictKey:
-      QUOTED_VALUE                               // 'hello world', "hello world"
+      quotedValue
```

Comment:
This is adding interpolations to dictKeys, which I think were not possible before.

Reply:
*Quoted* interpolations -- these are already possible since a `QUOTED_VALUE` can contain interpolations, e.g. in current master:
```python
cfg = OmegaConf.create({"x": "${oc.create:{'${key}': 0}}", "key": "foo"})
cfg.x  # {'foo': 0}
```
(it's possible it might not be explicitly tested, I'll double check later)

Reply:
Okay. I am having a hard time seeing a use case for this.
What the purpose of a key that changes dynamically based on some other config values?

Reply:
> What the purpose of a key that changes dynamically based on some other config values?

Last time you asked me this question my example didn't convince you so I won't try again :P
But at this point it would be more difficult (and make the code more complex) to remove this feature than to keep it, so do we really care? (unless we remove quoted values from valid dictionary keys altogether, but this seems a bit extreme since it would seriously limit the strings we can use as keys)

Reply:
Yeah, I thought we removed it last time.
In fact, I was surprised to see if even support quoted strings as dict keys.
This can very easily conflict with Structured Configs where keys must be legal Python identifiers (if we try to somehow override it with those dicts).

Now that quoted string is no longer a simple token I am even less excited about it being supported as a dict key.

(Quoted strings as dict keys are supported in Hydra as well (I think you added it when you aligned it with the direction the OmegaConf grammar was taking), but I doubt anyone is actually depending on it).

I don't mind keeping it, but I am a bit worried that this is something no one will use and is just adding complexity.

Reply:
As you wish -- it's easy enough to remove. Just a reminder though that we're talking about regular Python dictionaries here, that aren't necessarily going to be converted to DictConfig. Given the new powerful resolver features added in 2.1, it's hard to say what kind of new use cases will pop up depending on what people try to do with them...

Reply:
Okay then, let's remove it.

> Given the new powerful resolver features added in 2.1, it's hard to say what kind of new use cases will pop up depending on what people try to do with them...

Every language got some limitations.
It's easier to introduce it again if there is a strong use case than to take it away once a few users start to depend on it if it turns out that the complexity it adds is more than we bargained for.

Reply:
> Okay then, let's remove it.

Done in 02d7186